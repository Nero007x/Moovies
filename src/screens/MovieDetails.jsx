import React from 'react';
import { Text, Image, StyleSheet, ScrollView } from 'react-native';

const MovieDetails = ({ route }) => {
  const { movie } = route.params;

  return (
    <ScrollView style={styles.container}>
      <Image
        source={{ uri: `https://image.tmdb.org/t/p/w500${movie.poster_path}` }}
        style={styles.poster}
      />
      <Text style={styles.title}>{movie.title}</Text>
      <Text style={styles.overview}>{movie.overview}</Text>
      <Text style={styles.releaseDate}>Release Date: {movie.release_date}</Text>
      <Text style={styles.vote}>Rating: {movie.vote_average} / 10</Text>
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 10,
  },
  poster: {
    width: '100%',
    height: 400,
    resizeMode: 'contain',
  },
  title: {
    fontSize: 24,
    fontWeight: 'bold',
    marginVertical: 10,
  },
  overview: {
    fontSize: 16,
    marginVertical: 10,
  },
  releaseDate: {
    fontSize: 16,
    marginVertical: 5,
  },
  vote: {
    fontSize: 16,
    marginVertical: 5,
  },
});

export default MovieDetails;
