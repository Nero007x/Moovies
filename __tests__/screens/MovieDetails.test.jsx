import React from 'react';
import renderer from 'react-test-renderer';
import MovieDetails from '../../src/screens/MovieDetails'; // Adjust path as needed

const mockMovie = {
  title: 'Test Movie',
  overview: 'This is a test movie overview.',
  release_date: '2024-06-12',
  vote_average: 8.5,
  poster_path: '/poster.jpg'
};

it('renders MovieDetails correctly with sample data', () => {
  const tree = renderer.create(
    <MovieDetails
      route={{ 
        params: { movie: mockMovie }
      }}
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
