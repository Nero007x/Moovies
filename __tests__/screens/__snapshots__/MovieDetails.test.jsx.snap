// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MovieDetails correctly with sample data 1`] = `
<RCTScrollView
  style={
    {
      "flex": 1,
      "padding": 10,
    }
  }
>
  <View>
    <Image
      source={
        {
          "uri": "https://image.tmdb.org/t/p/w500/poster.jpg",
        }
      }
      style={
        {
          "height": 400,
          "resizeMode": "contain",
          "width": "100%",
        }
      }
    />
    <Text
      style={
        {
          "fontSize": 24,
          "fontWeight": "bold",
          "marginVertical": 10,
        }
      }
    >
      Test Movie
    </Text>
    <Text
      style={
        {
          "fontSize": 16,
          "marginVertical": 10,
        }
      }
    >
      This is a test movie overview.
    </Text>
    <Text
      style={
        {
          "fontSize": 16,
          "marginVertical": 5,
        }
      }
    >
      Release Date: 
      2024-06-12
    </Text>
    <Text
      style={
        {
          "fontSize": 16,
          "marginVertical": 5,
        }
      }
    >
      Rating: 
      8.5
       / 10
    </Text>
  </View>
</RCTScrollView>
`;
